### ISSUE TITLE
Enforce locale for commands run by tmt where error messages are parsed

### ISSUE THREAD

**Issue by happz** [teemtee/tmt#2082]
E.g. SSH-backed guests may fail to execute a given command, and report "Please login as the user" - without the correct locale, the language of the error message may change.

**Comment by happz** [comment]
> Looks we should add these environment variables in the following to ssh command, e.g.
> 
> ```shell
> LANG=en_US.UTF-8
> LC_ALL=en_US.UTF-8
> ```
> 
> @happz, what do you think?

Sounds like an obvious choice, but I have *zero* idea what might go wrong :) I had no chance to actually investigate the problem deeper, feel free to experiment.

**Comment by thrix** [comment]
Forcing some locale might break in some cases afaik, I would not rely on it.

**Comment by happz** [comment]
True, but the goal here is to make commands *tmt* runs for its own needs more predictable. Commands like `rsync` in `push()` and `pull()` or `git` when inspecting a repository with plans and tests, but definitely not those spawned by actual tests. Something like `LANG= LC_ALL= ssh ... -o "-LC_* -LANG_*" tmt-test-wrapper.sh`, i.e. set the locale for ssh itself, but not leak it into the environment of the test.
